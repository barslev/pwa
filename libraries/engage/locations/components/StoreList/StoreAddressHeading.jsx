// @flow
import React from 'react';
import StoreAddressDistance from './StoreAddressDistance';
import { storeHeading, storeName } from './style';

type Props = {
  store: Object;
}

/**
 * Renders a single store address heading.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const StoreAddressHeading = ({ store }: Props) => {
  const { name, distance, unitSystem } = store;

  return (
    <div className={storeHeading}>
      <div className={storeName} data-test-id="store-name">{name}</div>
      <StoreAddressDistance distance={distance} unitSystem={unitSystem} />
    </div>
  );
};

export default StoreAddressHeading;
