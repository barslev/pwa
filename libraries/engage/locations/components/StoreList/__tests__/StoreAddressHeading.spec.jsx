import React from 'react';
import { shallow } from 'enzyme';
import StoreAddressHeading from '../StoreAddressHeading';

jest.mock('../StoreAddressDistance', () => function StoreAddressDistance() {
  return null;
});

describe('<StoreAddressHeading>', () => {
  it('should render as expected', () => {
    const store = {
      name: 'ACME',
      distance: 5.3,
      unitSystem: 'system',
    };
    const wrapper = shallow((<StoreAddressHeading store={store} />));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div[data-test-id="store-name"]').text()).toEqual(store.name);
    expect(wrapper.find('StoreAddressDistance').props()).toEqual({
      distance: store.distance,
      unitSystem: store.unitSystem,
    });
  });
});
