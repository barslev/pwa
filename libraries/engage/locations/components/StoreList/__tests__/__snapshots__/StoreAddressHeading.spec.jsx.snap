// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreAddressHeading> should render as expected 1`] = `
<div
  className={
    Object {
      "data-css-1bvvjxu": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1qlb3ci": "",
      }
    }
    data-test-id="store-name"
  >
    ACME
  </div>
  <StoreAddressDistance
    distance={5.3}
    unitSystem="system"
  />
</div>
`;
